### PR TITLE
APM Enabled By Default

### PR DESCRIPTION
**Pull request by omerli**
### Preview link

https://docs-staging.datadoghq.com/omer.lifshitz/apm-enabled-by-default

### CODE REVIEW

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 1]
```diff
@@ -16,15 +16,15 @@ aliases:
   - /integrations/faq/using-rbac-permission-with-your-kubernetes-integration
 ---
 
-Take advantage of DaemonSets to deploy the Datadog Agent on all your nodes (or on specific nodes by [using nodeSelectors][1]).
+Take advantage of [DaemonSets][17] to deploy the Datadog Agent on all your nodes (or on specific nodes by [using nodeSelectors][1]).
 
 *If DaemonSets are not an option for your Kubernetes cluster, [install the Datadog Agent][2] as a deployment on each Kubernetes node.*
 
-## Configure RBAC permissions
+Setting up the agent as a DaemonSet requires 3 simple steps. If you run into issues throughout the process, make sure to check out the [troubleshooting section.][21]
```
Comment (cohenyair) [inline content/en/agent/kubernetes/daemonset_setup.md]: 3->three

Comment (omerli) [reply content/en/agent/kubernetes/daemonset_setup.md]:
```suggestion
Setting up the agent as a DaemonSet requires three simple steps. If you run into issues throughout the process, make sure to check out the [troubleshooting section.][21]
```

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 2]
```diff
@@ -194,6 +204,72 @@ For Agent v6.11+, the Datadog Agent can auto-detect the Kubernetes cluster name
...
+  name: traceport
+  protocol: TCP
+```
+
+Make sure that `DD_APM_ENABLED` is set to true:
+```yaml
+- {name: DD_APM_ENABLED, value: "true" }
+```
+
+**Important:** in order for the tracing libraries to find the agent, you need to add the following enviornement variable to every deployment file that belongs to an app you'd like to instrument:
```
Comment (hkaj) [inline content/en/agent/kubernetes/daemonset_setup.md]:
```suggestion
**Important:** in order for the tracing libraries to find the agent, you need to add the following environment variable to every deployment file that belongs to an app you'd like to instrument:
```

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 3]
```diff
@@ -194,6 +204,72 @@ For Agent v6.11+, the Datadog Agent can auto-detect the Kubernetes cluster name
...
+
+<pre>
+apiVersion: apps/v1
+kind: Deployment
+...
+    spec:
+      containers:
+      - name: <CONTAINER_NAME>
+        image: <CONTAINER_IMAGE>/<TAG>
+      <b>env:
```
Comment (hkaj) [inline content/en/agent/kubernetes/daemonset_setup.md]: the formatting is off

image

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 4]
```diff
@@ -194,6 +204,72 @@ For Agent v6.11+, the Datadog Agent can auto-detect the Kubernetes cluster name
...
+</pre>
+
+To make sure the flag is set properly in the app you'd like to deploy, test it with one of your applications pods:
+```shell
+kubectl exec <YOUR_APP_POD_NAME> env | grep DD_AGENT_HOST
+```
+
+DD_AGENT_HOST should be present and point to its parent node ip address.
+
+Lastly, for the changes to take effect, apply the configuration for the changes to take effect:
```
Comment (hkaj) [inline content/en/agent/kubernetes/daemonset_setup.md]:
```suggestion
Lastly, for the changes to take effect, apply the configuration: 
```

File: content/en/agent/kubernetes/daemonset_setup.md [hunk 5]
```diff
@@ -194,6 +204,72 @@ For Agent v6.11+, the Datadog Agent can auto-detect the Kubernetes cluster name
...
+<pre>
+apiVersion: apps/v1
+kind: Deployment
+...
+    spec:
+      containers:
+      - name: <CONTAINER_NAME>
+        image: <CONTAINER_IMAGE>/<TAG>
+      <b>env:
+          - name: DD_AGENT_HOST
```
Comment (hkaj) [inline content/en/agent/kubernetes/daemonset_setup.md]: I see you removed tracer configs. Can you confirm that all tracers lookup this env var? I'm mostly worried about existing deployments that want to migrate to k8s but won't upgrade tracers (so if the change was introduced recently that won't cover them).

File: content/en/agent/kubernetes/helm.md [hunk 6]
```diff
@@ -107,27 +62,42 @@ tiller-deploy-f54b67464-jl5gm 1/1 Running 0 3h16m
...
 {{% /tab %}}
+{{< /tabs >}}
 
-{{% tab "Helm v3+" %}}
+This chart adds the Datadog Agent to all nodes in your cluster via a DaemonSet. It also optionally deploys the [kube-state-metrics chart][5] and uses it as an additional source of metrics about the cluster. A few minutes after installation, Datadog begins to report hosts and metrics.
 
-```bash
-helm install <RELEASE_NAME> --set datadog.apiKey=<DATADOG_API_KEY> stable/datadog
+### Standard Install with APM and DogStatsD Enabled
+For a quick setup that will automatically enable APM and\or Custom DogStatsD metrics, you can use the following command:
```
Comment (hkaj) [inline content/en/agent/kubernetes/helm.md]:
```suggestion
For a quick setup that will automatically enable APM and/or Custom DogStatsD metrics, you can use the following command:
```

File: content/en/agent/kubernetes/helm.md [hunk 7]
```diff
@@ -107,27 +62,42 @@ tiller-deploy-f54b67464-jl5gm 1/1 Running 0 3h16m
...
+{{< tabs >}}
+{{% tab "Helm v3+" %}}
+```bash 
+helm install <RELEASE_NAME> --set datadog.apiKey=<DATADOG_API_KEY>,datadog.apm.enabled=true,datadog.statsd.enabled=true,datadog.clusterAgent.admissionController.enabled=true stable/datadog
```
Comment (hkaj) [inline content/en/agent/kubernetes/helm.md]: Can the admissionController be enabled by default without any other change? I feel like this should be a separate PR to make it incremental. This feels too big a change.

File: content/en/agent/kubernetes/helm.md [hunk 8]
```diff
@@ -107,27 +62,42 @@ tiller-deploy-f54b67464-jl5gm 1/1 Running 0 3h16m
...
+```
+{{% /tab %}}
+{{% tab "Helm v1/v2" %}}
+```shell
+helm install --name <RELEASE_NAME> --set datadog.apiKey=<DATADOG_API_KEY>,datadog.apm.enabled=true,datadog.statsd.enabled=true,datadog.clusterAgent.admissionController.enabled=true stable/datadog
```
Comment (hkaj) [inline content/en/agent/kubernetes/helm.md]: ditto

File: content/en/agent/kubernetes/helm.md [hunk 9]
```diff
@@ -107,27 +62,42 @@ tiller-deploy-f54b67464-jl5gm 1/1 Running 0 3h16m
...
 ```
-
 {{% /tab %}}
 {{< /tabs >}}
 
-This chart adds the Datadog Agent to all nodes in your cluster via a DaemonSet. It also optionally deploys the [kube-state-metrics chart][5] and uses it as an additional source of metrics about the cluster. A few minutes after installation, Datadog begins to report hosts and metrics.
+This command takes advantage of Admission Controller. To learn more about the advantages of using Admission Controllers, visit the following [page][15].
```
Comment (hkaj) [inline content/en/agent/kubernetes/helm.md]: ditto

File: content/en/agent/kubernetes/helm.md [hunk 10]
```diff
@@ -143,71 +113,93 @@ As a best practice, a YAML file that specifies the values for the chart paramete
...
-Update the `env` section of your application's manifest with the following:
+<pre>
+apiVersion: apps/v1
+kind: Deployment
+...
+    spec:
+      containers:
+      - name: <CONTAINER_NAME>
+        image: <CONTAINER_IMAGE>/<TAG>
+      <b>env:
```
Comment (hkaj) [inline content/en/agent/kubernetes/helm.md]: formatting is off

File: content/en/agent/kubernetes/helm.md [hunk 11]
```diff
@@ -243,6 +235,52 @@ datadog:
           port: "%%port_0%%"
 ```
 
+### Installing the Helm server (Tiller)
```
Comment (hkaj) [inline content/en/agent/kubernetes/helm.md]: This is for helm's doc, not ours. We shouldn't document this.

**Comment by hkaj** [comment]
Ah, you'll also want to rebase. There was a recent revamp of the install doc.
